net/tcp.ss

```diff
@@ -0,0 +1,94 @@
+
+(export tcp-listener?
+        tcp-listen
+        tcp-client-port?
+        tcp-connect
+        tcp-accept
+        tcp-close
+        try-tcp-connect
+        tcp-connect/retry-until-deadline)
+
+(import :gerbil/gambit/exceptions :gerbil/gambit/ports :gerbil/gambit/threads
+        :std/format :std/pregexp :std/sugar
+        :clan/base :clan/timestamp)
+
+;; --------------------------------------------------------
+
+;; Listeners and connections with SYN/ACC
+
+(def SYN 22)
+(def ACK 6)
+(def (syn/ack port)
```

Comment:
Why do we need this SYN/ACK protocol?

And why do you bake it in every tcp listener or client that uses this library?

It looks like you're re-doing in userland what the underlying TCP stack should already be doing for you at a lower-level. That's weird, makes this code incompatible with existing services, and is probably not necessary.

Reply:
I'm using this in order to know whether the connection has succeeded or not, mostly so that it can retry if it's still before the headline. Calling `open-tcp-client` doesn't always fail properly if there's no server ready yet, but the `syn-ack` does fail properly.

Reply:
Okay, I've changed this to use `u8-ready?` instead of `syn/ack`.